### PR TITLE
Use GitHub actions for CI

### PR DESCRIPTION
**Pull request by thomaslevesque**
This is basically the same as https://github.com/FakeItEasy/FakeItEasy/pull/1892, with the same commit structure.

I also included the same fixes as in https://github.com/FakeItEasy/FakeItEasy/pull/1895, and I updated FakeItEasy.Analyzers.Build to .NET 6 (so that we don't need .NET 5, and for consistency with FakeItEasy)

As mentioned in the chat, I also had to install .NET Core 2.1 on Windows. No idea why it wasn't necessary in FakeItEasy.

I updated how_to_build.md with the same changes as in FakeItEasy, but I noticed it's outdated. We'll need to update the build instructions, but I didn't want to do it in this PR.

Also, the AppVeyor build is failing, because we use an old VM image (VS 2019) which doesn't have .NET 6. But it's OK, since we're going to remove AppVeyor anyway...

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    needs: build
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: windows-artifacts
```
Comment (blairconrad) [inline .github/workflows/release.yml:33]: I was not expecting this, given we're on ubuntu

File: FakeItEasy.Analyzers.sln [hunk 2]
```diff
@@ -8,7 +8,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		.gitattributes = .gitattributes
 		.gitignore = .gitignore
 		.mailmap = .mailmap
-		appveyor.yml = appveyor.yml
```
Comment (blairconrad) [inline FakeItEasy.Analyzers.sln:11]: We did not add the `ci.yml`?

**Review by blairconrad — CHANGES_REQUESTED** [review]
Nice work. ~A couple of~ One minor comment~s~, if you're inclined to change.

Also, you talked about adding caching for the restore phase. That's not supposed to be here, right?

File: FakeItEasy.Analyzers.sln (hunk 2 above)
Comment (thomaslevesque) [reply FakeItEasy.Analyzers.sln:11]: because I forgot! Also didn't do it in FakeItEasy...

**Comment by thomaslevesque** [comment]
> Also, you talked about adding caching for the restore phase. That's not supposed to be here, right?

No, I didn't look into it yet

File: FakeItEasy.Analyzers.sln (hunk 2 above)
Comment (blairconrad) [reply FakeItEasy.Analyzers.sln:11]: I'm forever forgetting to add files to the sln. More trouble than it's worth, I sometimes think.

**Review by blairconrad — DISMISSED** [review]
Thanks, @thomaslevesque!

**Review by blairconrad — APPROVED** [review]
Smashing!

**Pull request merged**